### PR TITLE
Added ability to call analytics report manually

### PR DESCRIPTION
**Pull request by bsekachev**
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
    - Added manual initiation of analytics reports calculations.
    - Introduced new filter types and functions for analytics, quality reports, conflicts, and settings.
    - Added performance calculation functionalities in the API.
- **Enhancements**
    - Improved analytics report creation and listening functionalities.
    - Enhanced user interface for analytics pages, including new buttons and status indicators.
- **Bug Fixes**
    - Updated analytics report generation logic to handle missing IDs more gracefully.
- **Refactor**
    - Refactored permissions handling in analytics reports.
    - Organized import statements for better readability across several modules.
- **Style**
    - Updated styles for analytics components, improving layout and visual hierarchy.
- **Tests**
    - Enhanced Cypress tests to cover new functionalities and improved existing tests for better reliability.

### CODE REVIEW

**Comment by bsekachev** [comment]
@coderabbitai review

**Comment by nmanovic** [comment]
@coderabbitai review

File: cvat-ui/src/components/analytics-page/styles.scss [hunk 1]
```diff
@@ -16,12 +16,20 @@
...
 .cvat-task-quality-page,
+.cvat-analytics-overview,
 .cvat-project-quality-page {
     >.ant-row {
         margin-top: $grid-unit-size;
     }
 }
 
+.cvat-job-empty-performance-analytics-item {
+    > div > div {
```
Comment (klakhov) [inline cvat-ui/src/components/analytics-page/styles.scss:27]: Lets try to use specific class here

File: cvat-core/src/server-proxy.ts [hunk 2]
```diff
@@ -2411,6 +2414,92 @@ async function getAnalyticsReports(
...
+                    ...params,
+                }, { params: { rq_id: rqID } }).then((response) => {
+                    // todo: rewrite server logic, now it returns 202, 201 codes instead of RQ statuses
+                    if (response.status === 201) {
+                        listenerStorage[id].onUpdate.forEach((_onUpdate) => _onUpdate('finished', 0, 'Done'));
+                        resolve();
+                        return;
+                    }
+
+                    listenerStorage[id].onUpdate.forEach((_onUpdate) => _onUpdate('queued', 0, 'Analytics report is in progress'));
```
Comment (klakhov) [inline cvat-core/src/server-proxy.ts:2471]:
```suggestion
                    listenerStorage[id].onUpdate.forEach((_onUpdate) => _onUpdate('queued', 0, 'Analytics report calculation is in progress'));
```

File: cvat-core/src/server-proxy.ts [hunk 3]
```diff
@@ -2411,6 +2414,92 @@ async function getAnalyticsReports(
...
+                    setTimeout(checkStatus, 10000);
+                }).catch((errorData) => {
+                    if (!isNetworkError(errorData)) {
```
Comment (klakhov) [inline cvat-core/src/server-proxy.ts:2474]: Why do we need it? Should all long processes have such feature?

**Review by klakhov — COMMENTED** [review]
- Just noticed maybe its unrelated to the patch, but seems easy to fix. There is height jump while switching tabs:
height-jump

- While testing the patch the clickhouse was not up on my machine and the error occured. It seems full stack trace is shown right in the notification. I cant even scroll it :sweat_smile: No problem with this exact error but are all the possible errors are displayed this way?
image

What is the plan for this PR? Wait for requests PR or merge refactor later?

**Comment by bsekachev** [comment]
>  While testing the patch the clickhouse was not up on my machine and the error occured. It seems full stack trace is shown right in the notification. I cant even scroll it 😅 No problem with this exact error but are all the possible errors are displayed this way?

- Stacktraces should not be exposed to client
- Now we have several expressions, like `raise ValidationError(ex)`
- That is not related to this PR, but I will submit one more to fix it, also partially I aldready fixed it in another PR #7804, but perhaps  this case is not handled there likewise

File: cvat-core/src/server-proxy.ts (hunk 3 above)
Comment (bsekachev) [reply cvat-core/src/server-proxy.ts:2474]: I need to think one more time. 
I believe it is more UX (at least with specific number of retries before throwing the error and stopping asking the server).

But yes, it is better to send this change as separated PR.

**Comment by bsekachev** [comment]
> While testing the patch the clickhouse was not up on my machine and the error occured. It seems full stack trace is shown right in the notification. I cant even scroll it 😅 No problem with this exact error but are all the possible errors are displayed this way?

Checked one more time. 
It is a problem with code in endpoint POST /analytics/report?rq_id=<...>
Now the endpoint does not return enough data and will be deprecated/removed after the patch with requests is finished.

File: cvat-core/src/server-proxy.ts (hunk 3 above)
Comment (bsekachev) [reply cvat-core/src/server-proxy.ts]: Removed this

File: tests/cypress/e2e/features/analytics_pipeline.js [hunk 4]
```diff
@@ -194,35 +186,17 @@ context('Analytics pipeline', () => {
...
+                cy.get('button').contains('Request').click();
+            });
+
             checkCards(true);
             checkHistograms();
 
             cy.visit(`/tasks/${taskID}`);
             openAnalytics('task');
-            cy.wait('@getReport');
             checkCards(true);
```
Comment (klakhov) [inline tests/cypress/e2e/features/analytics_pipeline.js:200]: If we dont check empty cards with this function anymore, I suppose we can remove the `nonNull` paremeter from this function and call it just `checkCards()`

File: tests/cypress/e2e/features/analytics_pipeline.js [hunk 5]
```diff
@@ -194,35 +186,17 @@ context('Analytics pipeline', () => {
...
-            cy.login();
-            cy.intercept('GET', '/api/analytics/reports**').as('getReport');
-
             cy.visit(`/projects/${projectID}`);
             openAnalytics('project');
-            cy.wait('@getReport');
+            cy.get('.cvat-empty-performance-analytics-item').should('exist').and('be.visible').within(() => {
+                cy.get('button').contains('Request').click();
+            });
+
```
Comment (klakhov) [inline tests/cypress/e2e/features/analytics_pipeline.js:194]: Why dont we wait here for report request?  It can improve test stability

Comment (bsekachev) [reply tests/cypress/e2e/features/analytics_pipeline.js:194]: How will it impove test stability? 

`should('exist')` also will wait up to default timeout.

Comment (klakhov) [reply tests/cypress/e2e/features/analytics_pipeline.js:194]: On the second tought yes, default `sould` timeout should be enough

**Review by klakhov — APPROVED** [review]

**Pull request merged**